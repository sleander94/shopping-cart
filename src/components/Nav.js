import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/Nav.css';

const Nav = () => {
  return (
    <nav id="nav">
      <h1 className="title">Outdoor Store</h1>
      <div className="links">
        <Link to="/home">Home</Link>
        <Link to="/shop">Shop</Link>
        <Link to="/about">About</Link>
      </div>
    </nav>
  );
};

export default Nav;
