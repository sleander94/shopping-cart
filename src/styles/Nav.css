#nav {
  width: 100vw;
}

nav {
  display: grid;
  grid: 1fr / 3fr 1fr;
  align-items: center;
  padding: 24px 100px 24px 100px;
  width: 100vw;
  height: 15vh;
  min-height: 140px;
  background-color: black;
  color: white;
}

.title {
  font-size: 3rem;
  font-weight: 600;
  width: 400px;
}

.links {
  display: grid;
  grid-template-columns: 75px 75px 75px;
  align-items: center;
  gap: 12px;
}

.links a {
  color: white;
  font-size: 1.3rem;
  font-weight: 400;
  text-decoration: none;
  height: 1.4rem;
}

.links a:hover {
  font-size: 1.35rem;
}
