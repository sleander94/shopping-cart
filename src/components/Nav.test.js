import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import Nav from './Nav';

describe('Nav component', () => {
  it('renders correctly', () => {
    const nav = renderer.create(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    );
    expect(nav).toMatchSnapshot();
  });

  it('routes correct on click', () => {
    render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    );
    userEvent.click(screen.getByRole('link', { name: 'Home' }));
    expect(window.location.href).toContain('/home');
    userEvent.click(screen.getByRole('link', { name: 'Shop' }));
    expect(window.location.href).toContain('/shop');
    userEvent.click(screen.getByRole('link', { name: 'About' }));
    expect(window.location.href).toContain('/about');
    userEvent.click(screen.getByRole('link', { name: 'Outdoor Store' }));
    expect(window.location.href).toContain('/home');
  });
});
