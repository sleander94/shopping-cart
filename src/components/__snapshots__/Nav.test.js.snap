// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders correctly 1`] = `
<nav
  id="nav"
>
  <a
    className="title"
    href="/home"
    onClick={[Function]}
  >
    Outdoor Store
  </a>
  <div
    className="links"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/shop"
      onClick={[Function]}
    >
      Shop
    </a>
    <a
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </div>
</nav>
`;
